src/Adapters/Storage/Mongo/MongoTransform.js

```diff
@@ -508,7 +509,14 @@ function transformConstraint(constraint, inArray) {
   if (typeof constraint !== 'object' || !constraint) {
     return CannotTransform;
   }
-
+  const transformFunction = inArray ? transformInteriorAtom : transformTopLevelAtom;
+  const transformer = (atom) => {
+    const result = transformFunction(atom);
```

Comment:
no, `inArray`  is a boolean flag to know if we're expecting a transformation for an Array.

We have 2 transform methods (one for when the object is in an array `transformInteriorAtom` the other for top level `transformTopLevelAtom`)

L512 just help selecting the right function for the use case.